include/arch/arm/irq.h

```diff
@@ -136,6 +137,80 @@ static inline void arch_isr_direct_footer(int maybe_swap)
 	} \
 	static inline int name##_body(void)
 
+#if defined(CONFIG_DYNAMIC_DIRECT_INTERRUPTS)
+
+extern void z_arm_irq_direct_dynamic_dispatch_reschedule(void);
+
+/**
+ * @brief Macro to register an ISR Dispatcher (with re-scheduling request)
+ * for dynamic direct interrupts.
+ *
+ * This macro registers the ISR dispatcher function for dynamic direct
+ * interrupts, allowing the use of dynamic direct ISRs in the kernel.
+ * The dispatcher function is invoked when the direct
+ * interrupt occurs and then triggers the (soft) Interrupt Service Routine
+ * (ISR) that is registered dynamically (i.e. at run-time) into the IRQ table
+ * stored in SRAM. The ISR must be connected with irq_connect_dynamic() and
+ * enabled via irq_enable() before the dynamic direct interrupt can be
+ * serviced. This ISR dispatcher triggers thread re-secheduling upon return.
+ *
+ * Direct dynamic ISRs are designed for performance-critical interrupt handling
+ * and do not go through common interrupt handling code. They must be
+ * implemented in a way that it is safe to put them directly in the software
+ * vector table.
+ *
+ * @warning
+ * Although this routine is invoked at run-time, all of its arguments must be
+ * computable by the compiler at build time.
+ *
+ * @param irq_p IRQ line number.
+ * @param priority_p Interrupt priority.
+ * @param flags_p Architecture-specific IRQ configuration flags.
+ *
+ * Note: the function is an ARM Cortex-M only API.
+ */
+#define ARM_IRQ_DIRECT_DYNAMIC_CONNECT_RESCHEDULE(irq_p, priority_p, flags_p) \
```

Comment:
Can we have a single macro with an additional flag:

```
#define ARM_IRQ_DIRECT_DYNAMIC_CONNECT(irq_p, priority_p, flags_p, reschedule)
IRQ_DIRECT_CONNECT(irq_p, priority_p, \
		z_arm_irq_direct_dynamic_dispatch_reschedule, flags_p | (IRQ_FLAG_RESCHEDULE & reschedule))
```
or something similar?

Reply:
We can play with the pre-processor and the dispatcher name, stgh like:

```
#define ARM_IRQ_DIRECT_DYNAMIC_CONNECT(irq_p, priority_p, flags_p, reschedule_request)
IRQ_DIRECT_CONNECT(irq_p, priority_p, \
		z_arm_irq_direct_dynamic_dispatch_##reschedule_request, flags_p | (IRQ_FLAG_RESCHEDULE & reschedule))
```

But then we should ensure the macro to be used with certain arguments.
Not sure if it makes it look nicer at the end, @carlescufi

Reply:
Isn't there a way to check the flags at runtime from a single dispatch function?
If there isn't I still think your proposal above is better than documenting 2 separate macros.

Reply:
Eventually, we need to define at compile-time two dispatch functions returning true or false.
So there's no way around that.

OK, I can play with the pre-processor and document one single macro. Will try this and update the PR tomorrow. But the limitation is that the user then _must_ use the macro with certain _string_ arguments, otherwise there will be compilation errors. I don't know if we want that.

Reply:
done @carlescufi